### PR TITLE
[KYUUBI #3504] Extend JDBC URL to support catalog

### LINKED ISSUE #3504: [Improvement] Extend JDBC URL to support catalog
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

The current JDBC URL pattern is 
```
jdbc:kyuubi://<host>:<port>/<dbName>;<sessionVars>?<kyuubiConfs>#<engineVars>
```
As Kyuubi supports the "catalog" concept, we should extend the current JDBC URL pattern to support it.

### How should we improve?

Option 1:

Trino JDBC URL pattern[1] supports catalog and database, we can extend Kyuubi JDBC URL pattern to

```
jdbc:kyuubi://<host>:<port>/<catalogName>/<dbName>;<sessionVars>?<kyuubiConfs>#<engineVars>
```

For compatibility, if only one part is supplied, we should treat it as `dbName` rather than `catalogName`
```
jdbc:kyuubi://<host>:<port>/<dbName>;<sessionVars>?<kyuubiConfs>#<engineVars>
```

Option 2:

Add a new session variable "catalog" in `<sessionVars>` part, e.g.
```
jdbc:kyuubi://localhost:10009/sf1;catalog=tpcds
```

PostgreSQL uses a similar pattern(PostgreSQL uses `<database>.<schema>.<table>` to locate table)

```
jdbc:postgresql://localhost:5432/<database>?currentSchema=<schema>
```

[1] https://trino.io/docs/current/client/jdbc.html#connecting
[2] https://jdbc.postgresql.org/documentation/use/

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #3504 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java [hunk 1]
```diff
@@ -706,7 +706,11 @@ private void openSession() throws SQLException {
       openConf.put("set:hivevar:" + hiveVar.getKey(), hiveVar.getValue());
     }
     // switch the database
-    openConf.put("use:database", connParams.getDbName());
+    if (connParams.getCatalogName() != null) {
+      openConf.put("use:database", connParams.getCatalogName() + "." + connParams.getDbName());
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java:710]: we may want something like "use:catalog" here

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 2]
```diff
@@ -74,8 +71,13 @@ class TrinoSessionImpl(
     val sessionConf = sessionManager.getConf
     val connectionUrl = sessionConf.get(KyuubiConf.ENGINE_TRINO_CONNECTION_URL).getOrElse(
       throw KyuubiSQLException("Trino server url can not be null!"))
-    val catalog = sessionConf.get(KyuubiConf.ENGINE_TRINO_CONNECTION_CATALOG).getOrElse(
-      throw KyuubiSQLException("Trino default catalog can not be null!"))
+
+    val catalog = conf.getOrElse(
```
Comment (pan3793) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:75]: process at line 55 ?

**Comment by jiaoqingbo** [comment]
The failed test seems not related to this pr

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 3]
```diff
@@ -74,8 +73,13 @@ class TrinoSessionImpl(
     val sessionConf = sessionManager.getConf
     val connectionUrl = sessionConf.get(KyuubiConf.ENGINE_TRINO_CONNECTION_URL).getOrElse(
       throw KyuubiSQLException("Trino server url can not be null!"))
-    val catalog = sessionConf.get(KyuubiConf.ENGINE_TRINO_CONNECTION_CATALOG).getOrElse(
-      throw KyuubiSQLException("Trino default catalog can not be null!"))
+
+    if (catalogName.isEmpty) {
```
Comment (pan3793) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:77]: actually, `catalog` and `schema` are two similar variables, can we process them by unified style?

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala [hunk 4]
```diff
@@ -51,6 +51,13 @@ class SparkSessionImpl(
 
   override def open(): Unit = {
     normalizedConf.foreach {
+      case ("use:catalog", catalog) =>
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala:54]: comment at line 68 is outdated

**Comment by jiaoqingbo** [comment]
cc @pan3793， Please help take a look at this pr when you have time

**Review by pan3793 — APPROVED** [review]
LGTM, thanks

**Comment by pan3793** [comment]
Merging to master/1.6

**Pull request closed without merging**